buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/TestClustersPlugin.java

```diff
@@ -272,6 +283,54 @@ private void autoConfigureClusterDependencies(
         }));
     }
 
+    private static void configureCleanupHooks(Project project) {
+        synchronized (runningClusters) {
+            if (executorService == null || executorService.isTerminated()) {
+                executorService = Executors.newSingleThreadExecutor();
+            } else {
+                throw new IllegalStateException("Trying to configure executor service twice");
+            }
+        }
+        // When the Gradle daemon is used, it will interrupt all threads when the build concludes.
+        executorService.submit(() -> {
+            while (true) {
```

Comment:
NIT: the `while` loop is redundant when we sleep forever anyway I think :)

Reply:
I was thinking about that, but I think it conveys the intent better.

Reply:
sure, makes sense :)